solvers/NonlinearProgram.m

```diff
@@ -155,7 +155,6 @@
       obj.bbcon_lb = [];
       obj.bbcon_ub = [];
       
-      obj = obj.setSolver('default');
```

Comment:
I see where you're coming from. The point I'm making is a bit different though. By moving the `setSolver` call into `DirectTrajectoryOptimization` we'd no longer be calling it in any of the other classes that inherit from `NonlinearProgram` (eg. `InverseKinematics`). If we do want to move the `setSolver` call into `solve`, it should be `NonlinearProgram/solve` not `DirectTrajectoryOptimization/solve`.

Thanks for helping with our support for student snopt, by the way. Most of the Drake developers have regular snopt installed, which means the student version gets less testing.